File: message-bus/src/main/java/alma/icd/adapt/messagebus/AbstractRequestMessage.java
package alma.icd.adapt.messagebus;

/**
 * @author mchavan 21-Dec-2018
 */
public abstract class AbstractRequestMessage extends AbstractMessage implements RequestMessage {

	private String responseQueueName;
	
	@Override
	public String getResponseQueueName() {
		return responseQueueName;
	}

	@Override
	public void setResponseQueueName(String responseQueueName) {
		this.responseQueueName = responseQueueName;
	}

}

File: message-bus/src/main/python/adapt/messagebus/AbstractRequestMessage.py
from adapt.messagebus.Envelope import Envelope
from adapt.messagebus.RequestMessage import RequestMessage
from adapt.messagebus.AbstractMessage import AbstractMessage

class AbstractRequestMessage(AbstractMessage, RequestMessage):
    def __init__(self):
        super(AbstractMessage, self).__init__()
        self.__responseQueueName = None
    def getResponseQueueName(self):
        return self.__responseQueueName
    def setResponseQueueName(self, responseQueueName):
        self.__responseQueueName = responseQueueName

